### PR TITLE
Additional Guidelines for New Enablers

### PR DESCRIPTION
**Pull request by jason-fox**
* Split Checklist into three parts (New, Incubated, Mature)
* Added Submission Checklist
* Added `mkdocs serve` as start script (to facilitate local checking)

### CODE REVIEW

File: docs/GE_Checklist.md [hunk 1]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+    the code-base
+-   Basic documentation **MUST** exist in Markdown and **MUST** be complete
+    enough for usage of the component:
+    -   A Quick Start guide
+    -   Admin Installation instructions - how to create instances:
+        -   Standalone using configuration files
+        -   Using Docker/Docker-compose and injecting `ENV` variables
+    -   Developer Documentation - how to call the API programmatically
+    -   User Documentation - how to use the GUI
+    -   Apiary file (where necessary)
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: Apiary or OpenAPI? do we have a roadmap for the migration to OpenAPI?

File: docs/GE_Checklist.md [hunk 2]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
--   New entry in the [FIWARE Catalogue](https://catalogue-server.fiware.org)
-    under the corresponding Chapter agreed at TSC Level. See also
+### New Generic Enablers
+
+The following mimimal requirements must be met before a new contribution can be
+considered as an Incubated Generic Enabler.
+
+-   All New Generic Enablers **MUST** be presented to the TSC for candidature
+    and accepted into an existing Chapter
+-   The codebase **MUST** be available on GitHub
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: link to Github?

File: docs/GE_Checklist.md [hunk 3]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
 Any doubt or request to the FIWARE Foundation shall be made by sending an email
 to [fiware-help@lists.fiware.org](mailto:fiware-help@lists.fiware.org)
 
--   New entry in the [FIWARE Catalogue](https://catalogue-server.fiware.org)
-    under the corresponding Chapter agreed at TSC Level. See also
+### New Generic Enablers
+
+The following mimimal requirements must be met before a new contribution can be
+considered as an Incubated Generic Enabler.
+
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: a Generic Enabler must show how it integrates within the overall FIWARE Architecture

File: docs/GE_Checklist.md [hunk 4]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+-   All New Generic Enablers **MUST** be presented to the TSC for candidature
+    and accepted into an existing Chapter
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: the acceptance can take time, for instance to allow studying the quality of the product, internal architecture, software design, code cleanliness etc.

File: docs/GE_Checklist.md [hunk 5]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+-   The codebase **MUST** be available on GitHub
+-   The GitHub repository **MUST** include an appropriate Open Source License
+-   For functional readiness at a minimum, a new incubated enabler **MUST** be
+    able to be instantiated directly in a running state by any competent
+    developer without the need to refer to other dependent technologies - for
+    example:
+
+    -   Keyrock relies on MySQL, but a working MySQL can be easily instantiated
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: link to KeyRock?

File: docs/GE_Checklist.md [hunk 6]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+        by Docker and minimal data is seeded by Keyrock - the database volume
+        can be persisted. Knowledge of SQL is unnecessary for use.
+    -   An IoT Agent can have a dependency on MongoDB - the volumes are exposed
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: link  ...

File: docs/GE_Checklist.md [hunk 7]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+        so that the data will be persisted if the Docker container goes down.
+        Alternatively an IoT Agent can be memory based, but all configuration
+        can be set up using a config file or Docker params and then data
+        injected by HTTP requests. An understanding of MongoDB is unnecessary to
+        get it to work.
+    -   Quantum Leap works with Grafana and Crate-DB - appropriate set-up and
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: link

File: docs/GE_Checklist.md [hunk 8]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
+    -   A Quick Start guide
+    -   Admin Installation instructions - how to create instances:
+        -   Standalone using configuration files
+        -   Using Docker/Docker-compose and injecting `ENV` variables
+    -   Developer Documentation - how to call the API programmatically
+    -   User Documentation - how to use the GUI
+    -   Apiary file (where necessary)
+-   For a stateful component, the state **MUST** be persistable between
+    instantiations, additional manual set-up **MUST** not be required.
+-   There **MUST** be a commitment that the team behind the product will follow
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: how that commitment is expressed?

What happens if the commitment is not finally fulfilled? 

How the incubation process terminates? 

Can the incubation process terminate by rejecting the Enabler? What criteria are used for doing so? 

What criteria are used for transitioning to a full Enabler? Should we ask for some numbers on adoption and community creation around the component?

File: docs/GE_Checklist.md [hunk 9]
```diff
@@ -6,40 +6,138 @@ to be taken in order to publish a new FIWARE GE.
...
--   Provide access to the corresponding information in order to see your
-    component reflected in the [ monitoring
+-   Access **MUST** be provided to the corresponding information in order to see
+    your component reflected in the [ monitoring
     tool](https://docs.google.com/spreadsheets/d/1yyZNUlAPDcqjnD-gIoGOd5SZfVDJXO36G75xTDL0HgA/edit#gid=0
     FIWARE - TSC Enablers Dashboard) (Ask FF Staff)
+
+It is accepted that Incubated Enablers may not currently comply with all
+"**MUST**" requirements of the [Development Guidelines](development.md), but are
+expected to be working towards fulfilling them.
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: in what timeframe?

File: package.json [hunk 10]
```diff
@@ -33,6 +33,7 @@
 		"node": ">=8.6"
 	},
 	"scripts": {
+		"start" : "mkdocs serve",
```
Comment (jmcanterafonseca) [inline package.json]: is this really part of this PR or should it be moved to another PR?

File: docs/GE_Checklist.md (hunk 2 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 9e99c49

File: docs/GE_Checklist.md (hunk 5 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 9e99c49

File: docs/GE_Checklist.md (hunk 6 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 9e99c49

File: docs/GE_Checklist.md (hunk 7 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 9e99c49 - link added.

File: docs/GE_Checklist.md (hunk 8 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed : 9e99c49 - Text expanded. 

The testing of the commitment  should be part of standard QA testing (as it is for all other enablers)

Failure to comply will eventually result in the same sanction as a full member that does not comply - the Component will be quarantined and eventually removed. The only difference is that an Incubated Enabler has more leeway and time to come up to standard.

Successful progression to full member should be rubber-stamped by the TSC.

File: docs/GE_Checklist.md (hunk 3 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 7228c75 - Additional bullet added

File: docs/GE_Checklist.md (hunk 4 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: Fixed 866fc44d1 - added acceptance text

File: docs/GE_Checklist.md (hunk 1 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: NTC - Previous requirements stated Apiary. All existing APIs should therefore already have an Apiary file. Tools exist to programmatically convert from Apiary to OpenAPI if necessary.  Changing this requirement should be a separate issue if required - this PR just moved an existing line of text.

File: docs/GE_Checklist.md (hunk 9 above)
Comment (jason-fox) [reply docs/GE_Checklist.md]: NTC - Left vague as it will depend on the enabler. Some may need more time than others.

File: package.json (hunk 10 above)
Comment (jason-fox) [reply package.json]: NTC - already described as part of the PR - It is useful and don't see the need to raise a separate issue for a one-liner like this.

File: docs/GE_Checklist.md [hunk 11]
```diff
@@ -6,40 +6,151 @@ to be taken in order to publish a new FIWARE GE.
...
+        if the Docker container goes down. Alternatively an IoT Agent can be
+        memory based, but all configuration can be set up using a config file or
+        Docker params and then data injected by HTTP requests. An understanding
+        of MongoDB is unnecessary to get it to work.
+    -   [QuantumLeap](https://quantumleap.rtfd.io/) works with Grafana and
+        Crate-DB - appropriate set-up and docker compose are available in the
+        Quick Start. Quantum Leap can be used without in depth knowledge of
+        Crate-DB
+
+-   For all webservice components, a `Dockerfile` **MUST** be available within
```
Comment (jmcanterafonseca) [inline docs/GE_Checklist.md]: a link from `Dockerfile` to https://docs.docker.com/engine/reference/builder/ would be nice

File: package.json (hunk 10 above)
Comment (jmcanterafonseca) [reply package.json]: we can leave as it is, but in the future one issue, one PR, please, even if they are one liners :)

thanks!
